src/folding/circuits/sum_check.rs

```diff
@@ -63,25 +72,90 @@ impl<F: PrimeField> DensePolynomialVar<F> {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct IOPProofVar<C: CurveGroup> {
+    // We have to be generic over a CurveGroup because instantiating a IOPProofVar will call IOPSumCheck which requires a CurveGroup
+    pub proofs: Vec<DensePolynomialVar<C::ScalarField>>,
+    pub claim: FpVar<C::ScalarField>,
+}
+
+impl<C: CurveGroup> AllocVar<IOPProof<C::ScalarField>, C::ScalarField> for IOPProofVar<C>
+where
+    <C as Group>::ScalarField: Absorb,
+{
+    fn new_variable<T: Borrow<IOPProof<C::ScalarField>>>(
+        _cs: impl Into<Namespace<C::ScalarField>>,
+        _f: impl FnOnce() -> Result<T, SynthesisError>,
+        _mode: AllocationMode,
+    ) -> Result<Self, SynthesisError> {
+        _f().and_then(|c| {
+            let cs = _cs.into();
+            let cp: &IOPProof<C::ScalarField> = c.borrow();
+            let claim = IOPSumCheck::<C, PoseidonTranscript<C>>::extract_sum(cp);
+            let claim = FpVar::<C::ScalarField>::new_variable(cs.clone(), || Ok(claim), _mode)?;
+            let mut proofs =
+                Vec::<DensePolynomialVar<C::ScalarField>>::with_capacity(cp.proofs.len());
+            for proof in cp.proofs.iter() {
+                let poly = DensePolynomial::from_coefficients_slice(&proof.coeffs);
+                let proof = DensePolynomialVar::<C::ScalarField>::new_variable(
+                    cs.clone(),
+                    || Ok(poly),
+                    _mode,
+                )?;
+                proofs.push(proof);
+            }
+            Ok(Self { proofs, claim })
+        })
+    }
+}
+
+/// Auxillary information for a polynomial, (num_variables, max_degree)
+type PolyAuxInfo = (usize, usize);
+
+#[derive(Clone, Debug)]
+pub struct PolyAuxInfoVar<F: PrimeField> {
+    pub num_variables: FpVar<F>,
+    pub max_degree: FpVar<F>,
+}
+
+impl<F: PrimeField> AllocVar<PolyAuxInfo, F> for PolyAuxInfoVar<F> {
+    fn new_variable<T: Borrow<PolyAuxInfo>>(
+        _cs: impl Into<Namespace<F>>,
+        _f: impl FnOnce() -> Result<T, SynthesisError>,
+        _mode: AllocationMode,
```

Comment:
Is there any reason to add `_` before `cs`, `f`, and `mode`? they are being used

Reply:
Woops seems like I got an autocomplete glitch. Fixed [here](https://github.com/privacy-scaling-explorations/folding-schemes/pull/47/commits/95a4f23733c296130f616f260c1b12d17d40474c). Thanks for the heads up!